pyexchange/coinbase.py

```diff
@@ -0,0 +1,295 @@
+
+    def get_all_trades(self, pair: str, page_number: int = 1) -> List[Trade]:
+        assert(isinstance(pair, str))
+        assert(isinstance(page_number, int))
+
+        limit = 100
+
+        result = self._http_unauthenticated("GET", f"/products/{pair}/trades?before={page_number}&limit={limit}", {})
+
+        return list(map(lambda item: Trade.from_all_list(pair, item), result))
+
+    def _http_authenticated(self, method: str, resource: str, body: dict):
+        assert(isinstance(method, str))
+        assert(isinstance(resource, str))
+        assert(isinstance(body, dict) or (body is None))
+
+        data = json.dumps(body, separators=(',', ':'))
+
+        timestamp = str(time.time())
+        message = ''.join([timestamp, method, resource, data or ''])
+        message = message.encode('ascii')
+        hmac_key = base64.b64decode(self.secret_key)
+        signature = hmac.new(hmac_key, message, hashlib.sha256)
+        signature_b64 = base64.b64encode(signature.digest()).decode('utf-8')
+
+        return self._result(requests.request(method=method,
+                                             url=f"{self.api_server}{resource}",
+                                             data=data,
+                                             headers={
+                                                 'Content-Type': 'Application/JSON',
+                                                 'CB-ACCESS-SIGN': signature_b64,
+                                                 'CB-ACCESS-TIMESTAMP': timestamp,
+                                                 'CB-ACCESS-KEY': self.api_key,
+                                                 'CB-ACCESS-PASSPHRASE': self.password
+                                             },
+                                             timeout=self.timeout))
+
+    def _http_unauthenticated(self, method: str, resource: str, body: dict):
+        assert(isinstance(method, str))
+        assert(isinstance(resource, str))
+        assert(isinstance(body, dict) or (body is None))
+
+        data = json.dumps(body, separators=(',', ':'))
+
+        return self._result(requests.request(method=method,
+                                             url=f"{self.api_server}{resource}",
+                                             data=data,
+                                             timeout=self.timeout))
+
+    def _result(self, result) -> Optional[dict]:
+        if not result.ok:
+            raise Exception(f"Coinbase API invalid HTTP response: {http_response_summary(result)}")
+
+        try:
+            data = result.json()
+        except Exception:
+            raise Exception(f"Coinbase API invalid JSON response: {http_response_summary(result)}")
+
+        return data
+
+    @staticmethod
+    def _get_precision(coin):
+        return {
+            'ETH': "%.7f",
+            'USDC': "%.2f"
+        }[coin]
+
+
+def http_response_summary(response) -> str:
```

Comment:
Why have you copied that method from `pymaker`...?

Reply:
oops, fixed with second commit